scala/scala.bzl

```diff
@@ -382,6 +382,13 @@ def collect_srcjars(targets):
             srcjars += [target.srcjars.srcjar]
     return srcjars
 
+def update_jars_to_labels(jars2labels, target, java_provider):
+  for jar in (java_provider.compile_jars + java_provider.transitive_runtime_jars):
+    if jar.path not in jars2labels:
```

Comment:
i guess this is to eliminate `d` showing up in both `c` and `d` right? if so maybe a named inner method, a comment or both would help clarify this